pdns/libssl.cc

```diff
@@ -288,7 +290,11 @@ int libssl_ticket_key_callback(SSL* /* s */, OpenSSLTLSTicketKeysRing& keyring,
   return 1;
 }
 
+#ifdef BORINGSSL_API_VERSION
+static int libssl_server_name_callback(SSL* ssl, int* al, void* arg)
+#else
```

Comment:
`int` is also the correct return value with OpenSSL, apparently. The compiler doesn't complain because the OpenSSL header helpfully casts the callback function to `(void (*)(void))cb` :facepalm: 
I think you can just change it from `long` to `int`.